### DISCUSSION TITLE
valueAsNumber apparently can be used for <select> as well

### DISCUSSION THREAD

**Discussion by hornta** [react-hook-form/react-hook-form#7695] [Q&A]
The documentation states that the property `valueAsNumber` which is provided for the register method is only applied to text inputs. But when I use it for a select input I can see that valueAsNumber are applied as well.

https://react-hook-form.com/api/useform/register/#registerRef

I believe either the documentation is lying or it's a bug but I couldn't find a matching issue template which is why I'm writing here.

**Comment by Moshyfawn** [comment]
Track here #7696

**Reply by Moshyfawn** [reply]
The issue has been moved to react-hook-form/documentation#722 as a docs enhancement rather than a RHF bug

**Comment by bluebill1049** [comment]
The intention is only made is work for text input (HTMLInputElement), HTMLSelectElement "works" but not supported or potential having issues. I would recommend to use `setValueAs` instead.
